nano/node/network.cpp

```diff
@@ -236,6 +236,28 @@ void nano::network::flood_message (nano::message const & message_a, bool const i
 	}
 }
 
+void nano::network::flood_vote (std::shared_ptr<nano::vote> vote_a, float scale)
+{
+	auto list (list_fanout (scale));
+	for (auto i (list.begin ()), n (list.end ()); i != n; ++i)
+	{
+		if (!node.rep_crawler.is_pr (**i))
```

Comment:
Do we risk not meeting the desired fanout because half of the list could be PRs? Not a problem when nodes >>PRs but for now?

Reply:
Yes, we can do this in list. I don’t usually like to pass booleans to inject control in the target function. We could have a similar function to list that is list_non_pr which, after shuffle, iterates from the beginning removing prs until it reaches the end of list or count and then resizes the deque

Reply:
Added, specifically in https://github.com/nanocurrency/nano-node/pull/2468/commits/60b2fe2b2a3adb7e7d5de0d0cdf046d2d4c398e4 . `fanout()` was extracted to avoid having both `list_fanout ()` and `list_non_pr_fanout ()`. The number of caller sites was sufficiently small that it seemed like a good idea, please take a look.

You mentioned a more complicated loop that would include PRs if the working list fell below count, is that still the intention? As it stands now it can't include PRs even if the count falls below the desired.